plugins/elantp/fu-elantp-i2c-device.c

```diff
@@ -44,12 +47,97 @@ fu_elantp_i2c_device_to_string(FuDevice *device, guint idt, GString *str)
 	fu_string_append_kx(str, idt, "IcPageCount", self->ic_page_count);
 	fu_string_append_kx(str, idt, "IapType", self->iap_type);
 	fu_string_append_kx(str, idt, "IapCtrl", self->iap_ctrl);
+	fu_string_append(str, idt, "BindPath", self->bind_path);
+	fu_string_append(str, idt, "BindId", self->bind_id);
+}
+
+static gboolean
+fu_elantp_i2c_device_writeln(const gchar *fn, const gchar *buf, GError **error)
+{
+	int fd;
+	g_autoptr(FuIOChannel) io = NULL;
+
+	fd = open(fn, O_WRONLY);
+	if (fd < 0) {
+		g_set_error(error, FWUPD_ERROR, FWUPD_ERROR_INVALID_FILE, "could not open %s", fn);
+		return FALSE;
+	}
+
+	io = fu_io_channel_unix_new(fd);
+	return fu_io_channel_write_raw(io,
+				       (const guint8 *)buf,
+				       strlen(buf),
+				       1000,
+				       FU_IO_CHANNEL_FLAG_NONE,
+				       error);
+}
+
+static gboolean
+fu_elantp_i2c_device_rebind_driver(FuElantpI2cDevice *self, GError **error)
+{
+	if (self->bind_path == NULL || self->bind_id == NULL) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_INVALID_FILE,
+			    "no Path or ID for rebind driver");
+		return FALSE;
+	}
+
+	if (!fu_elantp_i2c_device_writeln(g_build_filename(self->bind_path, "unbind", NULL),
+					  self->bind_id,
+					  error))
+		return FALSE;
+	if (!fu_elantp_i2c_device_writeln(g_build_filename(self->bind_path, "bind", NULL),
+					  self->bind_id,
+					  error))
+		return FALSE;
```

Comment:
can you use fu_device_unbind_driver() here?

Reply:
I'm not sure, maybe unbind can work, but elan_i2c is a builtin driver and can't be found in /sys/module/, it can't be bind by similar method
https://github.com/fwupd/fwupd/blob/main/libfwupdplugin/fu-udev-device.c#L730